src/main/java/com/mraof/minestuck/item/block/TransportalizerItem.java

```diff
@@ -22,12 +27,30 @@ public TransportalizerItem(Block blockIn, Properties builder)
 	@Override
 	protected boolean updateCustomBlockEntityTag(BlockPos pos, Level level, @Nullable Player player, ItemStack stack, BlockState state)
 	{
-		if(stack.hasCustomHoverName() && stack.getHoverName().getString().length() == 4)
+		BlockEntity be = level.getBlockEntity(pos);
+		if(stack.hasTag() && be instanceof TransportalizerBlockEntity transportalizer)
 		{
-			BlockEntity be = level.getBlockEntity(pos);
-			if(be instanceof TransportalizerBlockEntity transportalizer)
-				transportalizer.setId(stack.getHoverName().getString().toUpperCase());
+			transportalizer.setId(stack.getTag().getString("idString"), player);
```

Comment:
This will now not set the id if it is already in use. However I'd argue that it should behave as it did before: set the id regardless of if it is used, and set the transportalizer to not be active if the id is already used.

There would then be two different scenarios/functions:
- Setting the id from player input/packet
- Copying over the id from item nbt